packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2030,101 +2030,110 @@ mixin WidgetInspectorService {
         'result': result,
       });
     }
+
+    Map<String, Object>? additionalCallback(DiagnosticsNode node, InspectorSerializationDelegate delegate) {
+      final Object? value = node.value;
+      final RenderObject? renderObject = value is Element ? value.renderObject : null;
+      if (renderObject == null) {
+        return const <String, Object>{};
+      }
+      assert(delegate.addAdditionalPropertiesCallback == additionalCallback);
+      // The new delegate for collecting properties on `renderObject`.
+      // Setting `subtreeDepth` to 0 to skip child RenderObjects. Also removes
+      // the additional callback from `delegate` since we should only collect
+      // RenderObjects from the element tree.
+      final DiagnosticsSerializationDelegate newDelegate = InspectorSerializationDelegate(
+        groupName: delegate.groupName,
+        summaryTree: delegate.summaryTree,
+        maxDescendentsTruncatableNode: delegate.maxDescendentsTruncatableNode,
+        expandPropertyValues: delegate.expandPropertyValues,
+        subtreeDepth: 0,
+        includeProperties: true,
+        service: delegate.service,
+        addAdditionalPropertiesCallback: delegate.addAdditionalPropertiesCallback,
+      );
+
+      final Map<String, Object> additionalJson = <String, Object>{
+        'renderObject' : renderObject.toDiagnosticsNode().toJsonMap(newDelegate),
+      };
+
```

Comment:
Didn't touch the code below, moved the `addAdditionalPropertiesCallback` code to a named function so I can reference it.